### PR TITLE
Merge (Order, Categorical, Data_Labeler, DateTime) Profiles with Different Options

### PR DESCRIPTION
**Pull request by gme5078**
These profiles don't have options to merge, but the infrastructure is built out in this PR

### CODE REVIEW

File: dataprofiler/tests/profilers/test_data_labeler_column_profile.py [hunk 1]
```diff
@@ -241,3 +242,29 @@ def test_profile_merge(self, mock_instance):
...
+        self._setup_data_labeler_mock(mock_instance)
+
+        data = pd.Series(['1', '2', '3', '11'])
+        data2 = pd.Series(['4', '5', '6', '7', '9', '10', '12'])
+        options = DataLabelerOptions()
+        options.max_sample_size = 20
+        profiler = DataLabelerColumn(data.name, options=options)
+        profiler.update(data)
+
+        options2 = DataLabelerOptions()
```
Comment (AnhTruong) [inline dataprofiler/tests/profilers/test_data_labeler_column_profile.py:261]: can we add a test for two data labelers with different options?

**Review by AnhTruong — COMMENTED** [review]
some comments

File: dataprofiler/profilers/data_labeler_column_profile.py [hunk 2]
```diff
@@ -115,7 +113,13 @@ def __add__(self, other):
             raise TypeError("Unsupported operand type(s) for +: "
                             "'DataLabelerColumn' and '{}'".format(
                                 other.__class__.__name__))
-
+        
+        if self.data_labeler != other.data_labeler \
```
[the remark this answers is not in the document]
Comment (gme5078) [reply dataprofiler/profilers/data_labeler_column_profile.py:117]: yea, it checks the label mapping of the model

File: dataprofiler/tests/profilers/test_data_labeler_column_profile.py (hunk 1 above)
Comment (gme5078) [reply dataprofiler/tests/profilers/test_data_labeler_column_profile.py:262]: Done!

File: dataprofiler/profilers/categorical_column_profile.py [hunk 3]
```diff
@@ -23,13 +23,11 @@ def __init__(self, name, options=None):
         :param name: Name of data
         :type name: String
         """
-
-        self.options = options
+        if options and not isinstance(options, CategoricalOptions):
+            raise ValueError("options must be of type CategoricalOptions.")
```
Comment (lettergram) [inline dataprofiler/profilers/categorical_column_profile.py:27]: I don't think a user would understand this and is it even possible to error out?

File: dataprofiler/profilers/data_labeler_column_profile.py [hunk 4]
```diff
@@ -269,6 +276,7 @@ def update(self, df_series):
         df_series = df_series.sample(sample_size)
 
         profile = dict(sample_size=sample_size)
+        self._update_predictions(df_series, {}, profile)
```
Comment (lettergram) [inline dataprofiler/profilers/data_labeler_column_profile.py:279]: what's the 3rd dict? It's probably good to add the variable name

File: dataprofiler/profilers/datetime_column_profile.py [hunk 5]
```diff
@@ -44,15 +45,16 @@ def __init__(self, name, options=None):
         :param options: Options for the datetime column
         :type options: DateTimeOptions
         """
-        self.options = options
+        if options and not isinstance(options, DateTimeOptions):
+            raise ValueError("options must be of type DateTimeOptions.")
```
Comment (lettergram) [inline dataprofiler/profilers/datetime_column_profile.py:49]: Same comment as above, would the users understand this?

File: dataprofiler/profilers/order_column_profile.py [hunk 6]
```diff
@@ -17,13 +18,13 @@ def __init__(self, name, options=None):
         :param options: Options for the Order column
         :type options: OrderOptions
         """
+        if options and not isinstance(options, OrderOptions):
+            raise ValueError("options must be of type OrderOptions.")
```
Comment (lettergram) [inline dataprofiler/profilers/order_column_profile.py:22]: See above

**Review by AnhTruong — APPROVED** [review]

File: dataprofiler/profilers/categorical_column_profile.py (hunk 3 above)
Comment (gme5078) [reply dataprofiler/profilers/categorical_column_profile.py:27]: shouldn't happen, but who knows. How should I make this error message more clear? I think if a user gets this error, they will know what it means imo

**Review by AnhTruong — APPROVED** [review]

**Review by lettergram — APPROVED** [review]

**Pull request merged**